syntax = "proto3";

package matrices;

message MatrixRequest {
    int32 matrix_id = 1;
    repeated int32 row = 2;
    repeated int32 column = 3;
    int32 i = 4; 
    int32 j = 5;
}

message MatrixResponse {
    string message = 1;
    int32 result = 2;
}

message ResultRequest {
    int32 matrix_id = 1;
    int32 result = 2;
    int32 i = 3;
    int32 j = 4;
}

message ResultResponse {
    string message = 1;
}

// MatrixService Definition
service MatrixService {
    rpc DistributeMatrix (MatrixRequest) returns (MatrixResponse);
    rpc SaveResult (ResultRequest) returns (ResultResponse);
}